### PR TITLE
React 16.9 upgrade

### PR DESCRIPTION
**Pull request by cmdcolin**
This fixes the issues we have seen with the act function throwing wild warnings

This PR is still pending react 16.9 getting a stable release but I figured I'd open the PR

I might also try to do a couple things, for example simplify the fetch logic in UrlInput with a useFetch hook or similar

### CODE REVIEW

**Comment by cmdcolin** [comment]
The UrlInput (e.g. the useRef thing I thought would fix it) is still not working in this pr. We basically just need async form validation but it's been weird

File: .eslintrc.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = {
     'prettier/react',
     'prettier/@typescript-eslint',
   ],
-  plugins: ['@typescript-eslint', 'prettier'],
+  plugins: ['@typescript-eslint', 'prettier', 'react-hooks'],
```
Comment (garrettjstevens) [inline .eslintrc.js]: Do we need to enable the `react-hooks` plugin? I think `react-app` in `extends` takes care of enabling the hooks lint rules.

Comment (cmdcolin) [reply .eslintrc.js]: ya doesn't apparently need this

**Comment by cmdcolin** [comment]
This is sufficiently out of date with the current master as to become obsolete. We'll re-evaluate react 16.9 upgrade when it actually comes out...for now act warnings are squashed

**Pull request closed without merging**